linker/linker.cpp

```diff
@@ -649,6 +655,8 @@ class LoadTask {
     si_->load_bias = elf_reader.load_bias();
     si_->phnum = elf_reader.phdr_count();
     si_->phdr = elf_reader.loaded_phdr();
+    si_->rand_addr_segments = elf_reader.rand_addr_segments();
+    si_->sort_rand_addr_segments();
```

Comment:
Would not have this as a member function; instead, create copy, sort, and move-assign here.
I would rename `rand_addr_segments_by_<sort-key>` to make it super obvious it's sorted by XXX. Then add a comment at the variable definition that gives the rationale for why it's sorted by this key and not another (see me other comment).
Also declare the vector `const` since number and content of segment info does not change after copying.
Or maybe even use plain old C-array?